src/Ring-Core/RGBehavior.class.st

```diff
@@ -486,17 +503,16 @@ RGBehavior >> includesSelector: aString [
 ]
 
 { #category : #'testing - class hierarchy' }
-RGBehavior >> inheritsFrom: aClass [ 
-	"Answer whether the argument, aClass, is on the receiver's superclass 
-	chain."
+RGBehavior >> inheritsFrom: aClass [
+	"Answer whether the argument, aClass, is on the receiver's superclass chain."
 
 	| aSuperclass |
 	aSuperclass := self superclass.
-	[aSuperclass == nil]
-		whileFalse: 
-			[aSuperclass == aClass ifTrue: [^true].
-			aSuperclass := aSuperclass superclass].
-	^false
+	[ aSuperclass isNil ] whileFalse: [
+		aSuperclass == aClass ifTrue: [ ^ true ].
+		aSuperclass hasResolvedName ifFalse: [ ^ false ].
```

Comment:
Yes, good catch. Could we have a test for this?

Reply:
I mean, the case of iterating the hierarchy when a superclass is not resolved. That's an important ring case.